fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeActivity.java

```diff
@@ -149,97 +144,99 @@ public static HomeViewModel obtainViewModel(FragmentActivity activity) {
-          } else if (id == R.id.home_connect_menu) {
-            handleConnectNavigation();
-          } else if (id == R.id.home_launch_menu) {
-            handleLaunchNavigation();
-          } else if (id == R.id.home_wallet_menu) {
-            WalletFragment.openWallet(getSupportFragmentManager(), mViewModel.isWalletSetUp());
-          } else if (id == R.id.home_social_media_menu) {
-            handleSocialMediaNavigation();
-          }
-          return true;
+          HomeTab tab = HomeTab.findByMenu(item.getItemId());
+          boolean selected = openTab(tab);
+          if (selected) mViewModel.setCurrentTab(tab);
+          return selected;
         });
+    // Set an empty reselect listener to disable the onSelectListener when pressing multiple times
+    navbar.setOnItemReselectedListener(item -> {});
   }
 
-  private void handleSocialMediaNavigation() {
-    if (mViewModel.getLAOs().getValue() == null) {
-      ErrorUtils.logAndShow(
-          getApplicationContext(), TAG, new NoLAOException(), R.string.error_no_lao);
-      showHomeTab();
-    } else {
-      Log.d(HomeViewModel.TAG, "Opening social media activity");
-      startActivity(SocialMediaActivity.newIntent(this));
-    }
-  }
-
-  private void handleConnectNavigation() {
-    if (checkWalletInitialization()) {
-      openConnect();
-    } else {
-      showHomeTab();
+  /**
+   * Open the fragment based on the given tab and the application state
+   *
+   * @param tab to open
+   * @return true if the tab was opened and the menu should be selected on the navigation bar
+   */
+  private boolean openTab(HomeTab tab) {
+    switch (tab) {
+      case HOME:
+        setCurrentFragment(
+            getSupportFragmentManager(), R.id.fragment_home, HomeFragment::newInstance);
+        return true;
+      case CONNECT:
+        return openConnectTab();
+      case LAUNCH:
+        return openLaunchTab();
+      case WALLET:
+        WalletFragment.openWallet(getSupportFragmentManager(), mViewModel.isWalletSetUp());
+        return true;
+      case SOCIAL:
+        openSocialMediaTab();
+        return false;
+      default:
+        Log.w(TAG, "Unhandled tab type : " + tab);
+        return false;
     }
   }
 
-  private void handleLaunchNavigation() {
-    if (checkWalletInitialization()) {
-      setCurrentFragment(
-          getSupportFragmentManager(), R.id.fragment_launch, LaunchFragment::newInstance);
-    } else {
-      showHomeTab();
+  private boolean openConnectTab() {
+    // Check if the wallet is setup
+    if (!mViewModel.isWalletSetUp()) {
+      showWalletWarning();
+      return false;
     }
-  }
 
-  public void openConnect() {
-    if (checkSelfPermission(Manifest.permission.CAMERA) == PackageManager.PERMISSION_GRANTED) {
+    // Check for the permission, if it is not granted, ask for it
+    if (checkSelfPermission(Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED)
+      requestCameraPermission();
```

Comment:
I'm not sure if this is something that needs to be edited in the `.editorconfig`, but please always have curly brackets {} for if/else clauses. It prevents subtle mistakes.

Reply:
Probably yes, I'll look into that in another PR.